nixos/modules/installer/sd-card/sd-image.nix

```diff
@@ -115,6 +115,20 @@ in
       '';
     };
 
+    populateFirmware = mkOption {
+      type = types.path;
+      description = lib.mdDoc ''
+        Result of {option}`sdImage.populateFirmwareCommands`
+      '';
+      default = pkgs.runCommand "populateFirmware" {}
+      ''
+        mkdir firmware
+        ${config.sdImage.populateFirmwareCommands}
+        mkdir $out
+        mv firmware/* $out/
+      '';
+    };
```

Comment:
It's weirdly redundant to both have commands and a derivation for the firmware.

(Also the `default` here is missing a `defaultText`...)

`populateFirmwareCommands` is not typed, so already it's not being merged, I believe. So it's not like people would compose into it neatly.

We probably want to either:

 - Make `populateFirmwareCommands` always assume `pwd` is where files should go (thus we can use it in a subshell where we `cd` in the appropriate dir), and have `populateFirmwareScript` an output that is a `writeShellScript` holding the `populateFirmwareCommands`, even if a bit redundant.
 - Remove `populateFirmwareCommands` entirely and always rely on a `path`, and have the default `path` here be what previously `populateFirmwareCommands` held. In this case, we probably want to rename this `firmwarePartitionContents` or something along the line.

Reply:
I've opted for the second option and it pans out nicely I think.